app/src/main/java/com/example/tasky/agenda_details/presentation/utils/DateTimeHelper.kt

```diff
@@ -36,11 +39,17 @@ object DateTimeHelper {
         return duration.seconds
     }
 
-    fun getEpochSecondsFromDateAndTime(date: String, time: String): Long {
+    fun getEpochMillisecondsFromDateAndTime(date: String, time: String): Long {
         val dateFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy")
         val localTime = LocalTime.parse(time)
         val localDate = LocalDate.parse(date, dateFormatter)
         val zdt = ZonedDateTime.of(localDate, localTime, ZoneOffset.UTC)
-        return zdt.toEpochSecond()
+        return zdt.toEpochSecond() * 1000
+    }
+
+    fun formatEpochToDateString(epochMillis: Long): String {
+        val dateFormat = SimpleDateFormat("MMM d, HH:mm", Locale.getDefault())
```

Comment:
The more modern way to format dates is by using the Java LocalDateTime API, since `SimpleDateFormat` is not thread-safe (not an issue here, but as soon as you'd share this `sdf` instance across threads you can run into issues). This API is also quite simpler than what you have here to get this working with time zones and all:
```
val dateTime = ZonedDateTime.ofInstant(
    Instant.ofEpochMilli(millis),
    ZoneId.of("UTC")
)
val formattedDateTime = DateTimeFormatter
    .ofPattern("dd MMM yyyy")
    .format(dateTime)
```